src/app/core/oauth/oauth.service.ts

```diff
@@ -49,150 +54,78 @@ export class OauthService {
-      clientId: 'APP-MLXS7JVFJS9FEIFJ',
-      clientName: 'test',
-      clientEmailRequestReason: '',
-      memberName: 'asda',
-      redirectUrl: 'https://developers.google.com/oauthplayground?code=1R3eAY',
-      responseType: 'code',
-      stateParam: null,
-      userId: null,
-      userName: 'Leonardo Mendoza',
-      userOrcid: '0000-0002-2036-7905',
-      userEmail: null,
-      userGivenNames: null,
-      userFamilyNames: null,
-      nonce: null,
-      clientHavePersistentTokens: true,
-      scopesAsString: '/authenticate openid',
-    })
+    const value: OauthAuthorize = {
+      // tslint:disable-next-line: max-line-length
+      // TODO @angel please confirm that persistentTokenEnabled is always true https://github.com/ORCID/ORCID-Source/blob/master/orcid-web/src/main/webapp/static/javascript/ng1Orcid/app/modules/oauthAuthorization/oauthAuthorization.component.ts#L161
+      // TODO @angel by looking into analytics, I can see we have never reported a persistentTokenDisabled
+      persistentTokenEnabled: true,
+      // tslint:disable-next-line: max-line-length
+      // TODO @angel please confirm that  email access is always allowed know and at some point it was optional https://github.com/ORCID/ORCID-Source/blob/master/orcid-web/src/main/resources/freemarker/includes/oauth/scopes_ng2.ftl#L42
+      emailAccessAllowed: true,
+      approved,
+    }
+    return this._http
+      .post<RequestInfoForm>(
+        environment.BASE_URL + 'oauth/custom/authorize.json',
+        value,
+        { headers: this.headers }
+      )
+      .pipe(
+        retry(3),
+        catchError((error) => this._errorHandler.handleError(error)),
+        tap((requestInfo) => {
+          this.requestInfoSubject.next(requestInfo)
+        })
+      )
   }
 
   // call on by the OauthGuard
   // it send the Oauth query parameters to the backend and gets back the requestInfoForm
   // if the backend has an error declaring the Oauth parameters it will return a string on the errors array
 
   declareOauthSession(value: DeclareOauthSession): Observable<RequestInfoForm> {
-    // The oauth section is declared only one time when the user lands
     if (this.oauthSectionDeclared) {
       return this.requestInfoSubject
     }
 
-    // TODO @leomendoza123 @DanielPalafox remove mock data and use the following function
-
-    // return this._http
-    //   .post<RequestInfoForm>(
-    //     environment.BASE_URL + '/oauth/custom/declare.json',
-    //     value,
-    //     { headers: this.headers }
-    //   )
-    //   .pipe(
-    //     retry(3),
-    //     catchError((error) => this._errorHandler.handleError(error))
-    //   )
-    //   .pipe(
-    //     tap((data) => {
-    //       this.requestInfoSubject.next(data)
-    //       this.oauthSectionDeclared = true
-    //     })
-    //   )
+    return this._http
+      .post<RequestInfoForm>(
+        // tslint:disable-next-line:max-line-length
```

Comment:
@DanielPalafox please check this one should be on top of line 95 and 116

Reply:
Done